src/firecrest/status/health_check/health_checker.py

```diff
@@ -87,11 +87,16 @@ async def check(self) -> None:
             results = await asyncio.gather(*checks, return_exceptions=True)
             self.cluster.servicesHealth = results
         except Exception as ex:
-            error_message = f"{ex.__class__.__name__}"
+            error_message = (
+                f"HealthChecker execution failed with error: {ex.__class__.__name__}"
+            )
             if len(str(ex)) > 0:
-                error_message = f"{ex.__class__.__name__}: {str(ex)}"
+                error_message = f"HealthChecker execution failed with error: {ex.__class__.__name__} - {str(ex)}"
```

Comment:
If I understand correctly, now we'd have this error string for the user: `HealthChecker execution failed with error: ConnectTimeout`

I think this is now clear enough as a description of an error. We should describe what part of the infra is returning the error (ie, `Identity Provider ConnectTimeout` or `auth.cscs.ch ConnectTimeout` or `Obtaining Access Token ConnectTimeout`).

Maybe we should check per exception. I think that `ConnectTimeout` in our current code can only be triggered by this [line](https://github.com/eth-cscs/firecrest-v2/blob/Improved-HealtCheck-error-message/src/firecrest/status/health_check/health_checker.py#L50) by the `AsyncOAuth2Client` object.

Reply:
I was not convinced that adding more granular messages would help. This message is for the end-user not for the system admin that will manage FirecREST. 
Instead, the main fix here is the addition of this line ` raise ex` now, in case of exception the error will be properly logged. 
Meaning that end-user can communicate to the FirecREST owner that "HealthChecker execution failed" and sysadmin can now check in logs to find out what is happening.

Reply:
Ok, we can try with this and see how it works with this approach.
Thanks!